src/rest/CommunityEndpoints.js

```diff
@@ -1,24 +1,200 @@
+/**
+ * Streamr community product related functions
+ *
+ * Table of Contents:
+ *      admin: DEPLOY AND SETUP COMMUNITY   Functions for deploying the contract and adding secrets for smooth joining
+ *      member: JOIN & QUERY COMMUNITY      Publicly available info about communities and their members (with earnings and proofs)
+ *      member: WITHDRAW EARNINGS           Withdrawing functions, there's many: normal, agent, donate
+ *      admin: MANAGE COMMUNITY             Kick and add members
+ */
+
 import fetch from 'node-fetch'
 import {
     Contract,
     ContractFactory,
-    utils,
+    Wallet,
+    getDefaultProvider,
+    providers,
 } from 'ethers'
+import {
+    BigNumber,
+    computeAddress,
+    getAddress
+} from 'ethers/utils'
+import debugFactory from 'debug'
 
 import * as CommunityProduct from '../../contracts/CommunityProduct.json'
-import * as TestToken from '../../contracts/TestToken.json'
 
 import authFetch from './authFetch'
 
-export async function joinCommunity(communityAddress, memberAddress, secret = undefined) {
-    const url = `${this.options.restUrl}/communities/${communityAddress}/joinRequests`
+const debug = debugFactory('StreamrClient::CommunityEndpoints')
+
+/** @typedef {String} EthereumAddress */
+
+function throwIfBadAddress(address, variableDescription) {
+    try {
+        return getAddress(address)
+    } catch (e) {
+        throw new Error(`${variableDescription || 'Error'}: Bad Ethereum address ${address}`)
```

Comment:
if possible, should always check the signature of the error is what you expect before rethrowing.

The risk is that this reports "Bad Ethereum address" when the problem is something else entirely e.g. a `ReferenceError` and we've hidden the actual problem + stack trace and also misdirected the user to a non-problem.